test/conftest.py

```diff
@@ -69,17 +69,18 @@ def help():
 
 @pytest.fixture(scope='session', autouse=True)
 def filter_log():
-    # TODO: maybe we can use env variable to control whether doing this or not (this should be done)
     # A workround to use our custom Formatter in pytest.
     # Based on the discussion here 'https://github.com/pytest-dev/pytest/issues/2987'
     from snowflake.connector.secret_detector import SecretDetector
     import logging
     import pathlib
     # the directory of this conftest file
-    this_dir = pathlib.Path(__file__).parent.absolute()
+    this_dir = str(pathlib.Path(__file__).parent.absolute())
+    log_dir = os.getenv('CLIENT_LOG_DIR_PATH_DOCKER')
```

Comment:
It works the way it is, but if you do this you can get rid of the in-line if statement and the previous line too.

```suggestion
    log_dir = os.getenv('CLIENT_LOG_DIR_PATH_DOCKER', str(pathlib.Path(__file__).parent.absolute()))
```

Reply:
Perfect point!